challenges/2021-08-03-tracing-triangles/solutions/julia/ericwburden/ericwburden.jl

```diff
@@ -0,0 +1,38 @@
+module tracingtriangle # --------------------------------------------------------------
+
+export longest_path
+
+function longest_path(tree, row=1, col=1)
+    row == length(tree) && return tree[row][col]
+
+    left = longest_path(tree, row + 1, col)
+    right = longest_path(tree, row + 1, col + 1)
```

Comment:
Double recursion here, is there a big hit on this?

Just curious tbh.

Reply:
I figure it's pretty equivalent to something like

```
for x in list
  total += recursive_function(x)
end
```
where we know the list has length 2. It's definitely a brute-force, not super efficient approach. To make it efficient, you'd need to memoize on row/col.

Reply:
It's actually a pretty standard recursive form